src/pages/filter/ui/panels/function-panel/components/compound-request/compound-request.store.ts

```diff
@@ -0,0 +1,216 @@
+
+    const newRequestCondition = cloneDeep(this.requestCondition)
+
+    newRequestCondition.forEach((requestCondition, index) => {
+      if (index === activeRequestIndex) {
+        requestCondition[1] = newRequest
+      }
+    })
+
+    this.sendRequestAsync(newRequestCondition, resetName)
+  }
+
+  // add new request block
+  public handleRequestBlocksAmount(
+    type: string,
+    setActiveRequestIndex: (length: number) => void,
+    activeRequestIndex: number,
+  ): void {
+    if (type === 'ADD') {
+      const emptyBlock: [number, any] = [1, {}]
+      const newRequestCondition = [
+        ...cloneDeep(this.requestCondition),
+        emptyBlock,
+      ]
+
+      setActiveRequestIndex(newRequestCondition.length - 1)
+      this.sendRequestAsync(newRequestCondition, 'empty')
+    } else {
+      const newRequestCondition = cloneDeep(this.requestCondition).filter(
+        (_item, index) => index !== activeRequestIndex,
+      )
+
+      setActiveRequestIndex(newRequestCondition.length - 1)
+
+      this.sendRequestAsync(
+        newRequestCondition,
+        getResetType(newRequestCondition[newRequestCondition.length - 1][1]),
+      )
+    }
+  }
+
+  // change reuqest condition number
+  public handleRequestConditionNumber(
+    requestBlockIndex: number,
+    value: string,
+  ): void {
+    if (+value < 0) return
+
+    const newRequestCondition = cloneDeep(this.requestCondition)
+
+    newRequestCondition.map((item, index) => {
+      if (index === requestBlockIndex) {
+        item[0] = +value
+      }
+    })
+
+    this.sendRequestAsync(newRequestCondition)
+  }
+
+  // choose an active request block
+  public handleActiveRequest(
+    requestBlockIndex: number,
+    event: any,
+    setActiveRequestIndex: (index: number) => void,
+  ): void {
+    const classList = Array.from(event.target.classList)
+
+    const shouldMakeActive = classList.includes('step-content-area')
+
+    if (shouldMakeActive) {
+      setActiveRequestIndex(requestBlockIndex)
+
+      const currentRequest = this.requestCondition[requestBlockIndex]
+
+      this.setConditions({
+        resetName: getResetType(currentRequest[1]) || 'empty',
+      })
+    }
+  }
+
+  // cach values in every change
+  public setConditions({ newRequestCondition, resetName }: any): void {
+    functionPanelStore.setCachedValues<ICompoundRequestCachedValues>(
+      FuncStepTypesEnum.CompoundRequest,
+      {
+        conditions: {
+          approx: null,
+          state: null,
+          request: newRequestCondition || this.requestCondition,
+        },
+        reset: resetName || this.resetValue,
+      },
+    )
+  }
+
+  // root function to update data
+  public sendRequestAsync(
+    newRequestCondition: TRequestCondition[],
+    resetName?: string,
+  ): void {
+    const requestString = getFuncParams(FuncStepTypesEnum.CompoundRequest, {
+      request: newRequestCondition,
+    })
+      .slice(10)
```

Comment:
Why slice by 10? Can you create a well-named constant for it?
Or may be even utility function for slicing and removing spaces?